orbit/stan/dlt.stan

```diff
@@ -188,6 +205,20 @@ transformed parameters {
   } else {
     sea_sm = 0.0;
   }
+  
+  // needs to be above horseshoe prior setting
+  if (WITH_MCMC) {
+    // eqv. to obs_sigma ~ cauchy(SIGMA_EPS, CAUCHY_SD) T[SIGMA_EPS, ];
+    obs_sigma = SIGMA_EPS + CAUCHY_SD * tan(obs_sigma_unif_dummy[1]);
+  } else {
+    obs_sigma = obs_sigma_dummy[1];
+  }
+  
+  if (USE_HORSESHOE) {
+    tau = EXPECTED_SIZE / (NUM_OF_RR - EXPECTED_SIZE) * obs_sigma / sqrt(NUM_OF_OBS * GLOBAL_SHRINKAGE_FACTOR);
```

Comment:
noticed that `EXPECTED_SIZE` is set to be the number of regular regressors by default. Will this trigger an exception here (divided by 0)?

Reply:
yeah. it will if someone intentionally set prior to horseshoe and put zero number of regular regressors there.  i can fix that by putting a flag of USE_HORSESHOE = USE_HORSESHOE & (NUM_OF_RR >= 1)

Reply:
Adding @steveyang90 as reviewer since this involves quite some changes.